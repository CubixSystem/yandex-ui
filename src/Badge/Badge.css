.Badge {
    position: relative;

    display: inline-flex;

    vertical-align: middle;
}

.Badge-Dot {
    display: flex;
    justify-content: center;
    align-items: center;

    box-sizing: border-box;
    min-width: var(--badge-size-small);
    height: var(--badge-size-small);
    padding: 0 2px;

    font-family: var(--badge-font-family);
    font-size: var(--badge-font-size);
    font-weight: 400;
    line-height: 1;
    white-space: nowrap;

    color: var(--badge-typo-color);
    border: var(--badge-border-width) solid var(--badge-border-color);
    border-radius: var(--badge-border-radius);
    background-color: var(--badge-fill-color);
}

.Badge-Dot_withContent {
    min-width: var(--badge-size-large);
    height: var(--badge-size-large);
}

.Badge-Dot_withChildren {
    position: absolute;
    top: 0;
    right: 0;

    transform: translate(50%, -50%);
    transform-origin: 100% 0%;
}
