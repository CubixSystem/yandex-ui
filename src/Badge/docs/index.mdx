---
id: Badge
path: /components/badge

title: Badge
description: Индикатор-счетчик новой информации.

links:
 - label: source
   url: https://github.yandex-team.ru/search-interfaces/frontend/tree/master/packages/lego-components/src/Badge
 - label: startrek
   url: https://search.yandex-team.ru/stsearch?text=Badge.ts&facet.queue=ISL&facet.type=bug&facet.status=128
 - label: figma
   url: https://www.figma.com/file/CaoHFEO0w6PaE8cVaYFDoq8i/L-E-G-O?node-id=0%3A1
 - label: storybook
   url: https://lego.yandex-team.ru/components/lego/latest/?storybook=%2Fstory%2Fcontent-badge-desktop--playground

tabs:
 - Examples
 - Code
---
