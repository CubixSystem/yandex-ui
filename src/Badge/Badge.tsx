import React, { FC, RefObject, CSSProperties } from 'react';
import { cn } from '@bem-react/classname';

import './Badge.css';

export interface BadgeProps {
    /**
     * Ссылка на корневой DOM-элемент компонента
     */
    innerRef?: RefObject<HTMLDivElement>;

    /**
     * Пользовательские стили
     */
    style?: CSSProperties;

    /**
     * Дополнительный класс
     */
    className?: string;

    /**
     * Содержимое, отображаемое внутри значка
     */
    content?: string | number;

    /**
     * Цвет заливки блока
     */
    color?: string;

    /**
     * Цвет текста блока
     */
    textColor?: string;

    /**
     * Цвет обводки блока
     */
    outlineColor?: string;
}

export const cnBadge = cn('Badge');

/**
 * Индикатор-счётчик новой информации
 * @param {BadgeProps} props
 */
export const Badge: FC<BadgeProps> = ({
    color,
    textColor,
    outlineColor,
    content,
    children,
    innerRef,
    style,
    className,
    ...props
}) => (
    <div ref={innerRef} {...props} className={cnBadge(null, [className])}>
        {children}
        <div
            className={cnBadge('Dot', { withChildren: Boolean(children), withContent: Boolean(content) })}
            style={{ backgroundColor: color, color: textColor, borderColor: outlineColor, ...style }}
        >
            {content}
        </div>
    </div>
);

Badge.displayName = cnBadge();
