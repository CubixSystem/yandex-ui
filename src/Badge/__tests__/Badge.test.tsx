import React, { createRef } from 'react';

import { createClientRender } from '../../internal/testing';
import { Badge } from '../Badge';

describe('Badge', () => {
    const render = createClientRender();

    test('должен вернуть полный шаблон компонента (snapshot)', () => {
        const { container } = render(
            <Badge className="mix" content="1">
                <div />
            </Badge>,
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    test('должен быть установлен корректный displayName', () => {
        expect(Badge.displayName).toBe('Badge');
    });

    test('должен устанавливать ссылку на корневой DOM элемент', () => {
        const innerRef = createRef<HTMLDivElement>();
        render(<Badge innerRef={innerRef} />);

        expect(innerRef.current).not.toBe(null);
    });
});
