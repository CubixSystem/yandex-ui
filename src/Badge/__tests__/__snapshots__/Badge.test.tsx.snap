// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge должен вернуть полный шаблон компонента (snapshot) 1`] = `
<div
  class="Badge mix"
>
  <div />
  <div
    class="Badge-Dot Badge-Dot_withChildren Badge-Dot_withContent"
  >
    1
  </div>
</div>
`;
